navigator_helpers/tasks/sample_to_dataset/task_suite.py

```diff
@@ -895,3 +901,95 @@ def process_row(
             generated_data_w_seeds_df = pd.DataFrame()
 
         return generated_data_df, generated_data_w_seeds_df
+
+    def preprocess_sample_dataset(
+        self,
+        sample_dataset: pd.DataFrame,
+        system_prompt_type: str = "cognition",
+        num_samples: int = 25,
+    ) -> pd.DataFrame:
+        """
+        Preprocess the sample dataset by fixing the schema and sampling a limited number of records.
+
+        Args:
+            sample_dataset (pd.DataFrame): The sample dataset to preprocess.
+            system_prompt_type (str, optional): Type of system prompt to use. Defaults to 'cognition'.
+            num_samples (int, optional): Number of samples to keep. Defaults to 25.
+
+        Returns:
+            pd.DataFrame: A preprocessed DataFrame with fixed schema and limited samples.
+
+        Raises:
+            RuntimeError: If schema fixing fails after multiple attempts.
+        """
+        if num_samples > 50:
+            self.logger.warning(
+                "num_samples exceeds 50. Limiting to 50 random records."
+            )
+            num_samples = 50
+
+        self.logger.info(
+            f"  |-- 🪚 Trimming dataset to min(25, sample_size) = {min(sample_dataset.shape[0], 25)} records"
+        )
+        # If a big dataframe is passed, make sure to sample without replacement
+        if sample_dataset.shape[0] > num_samples:
+            sample_dataset = sample_dataset.sample(
+                n=num_samples, replace=False
+            ).reset_index(drop=True)
+
+        data_schema = list(sample_dataset.columns)
+        dataset_schema_preprocessing_prompt = (
+            DATASET_SCHEMA_PREPROCESSING_PROMPT_TEMPLATE.format(
+                sampled_dataset_column_list=str(data_schema),
```

Comment:
If the column name normalization became a standalone task, do you think we need the entire dataset or just the column names as input? I could imagine using examples from the dataset to ensure the column names make sense, but maybe that's getting ahead of ourselves (?)

Reply:
Yeah, this did occur to me if someone passes names that have no meaning, like column_1, column_2, etc. However, technically this is correct and proper formatting. I decided to keep it simple for now.